### PR TITLE
Snake? Snake!? SNAAAAKE! (adds snakes)

### PR DESCRIPTION
**Pull request by Togopalis**
:cl: Togopal
add: Snakes! They hunt vermin on the station!
add: They can be purchased in cargo.
/:cl:

[why]: # (Please add a short description [two lines down] of why you think these changes would benefit the game. If you can't justify it in words, it might not be worth adding.)

Mice can be a problem in late round random events, and I figured releasing a bunch of snakes in maintenance would be a more long-term solution than constantly running around with a crowbar when you hear them.

### CODE REVIEW

**Comment by Cyberboss** [comment]
@vuonojenmustaturska

File: code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		return TRUE
+	return FALSE
+
+/mob/living/simple_animal/hostile/poison/AttackingTarget()
+	if(is_type_in_typecache(target,edibles))
+		visible_message("[name] consumes [target] in a single gulp", "<span class='notice'>You consume [target] in a single gulp</span>")
+		QDEL_NULL(target) //Nom
+		adjustBruteLoss(-2)
+		return TRUE
+	else
```
Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/friendly/snake.dm]: Dangling else

**Comment by Armhulen** [comment]
snekmaster summoned

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by FantasticFwoosh** [comment]
I can't embed videos in comments.
https://youtu.be/P9yruQM1ggc

**Comment by MrDoomBringer** [comment]
>not providing images for @WJohn to consume

**Comment by Armhulen** [comment]
@MrDoomBringer uuuuuuh wjohn might be unhappy

**Comment by Togopalis** [comment]
The person who sprited it is sort of new to it sorry if it's sloppy compared to the other art but he's a god send to me because I can't sprite at all!!

**Comment by Armhulen** [comment]
they're codersprites

File: code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 1 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: how is this dangling wait what

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: it returns ..()

**Comment by FantasticFwoosh** [comment]
Im tired of all this 🅱️other🅱️u🅱️🅱️ing 🐍  on this 🅱️other🅱️u🅱️🅱️ing escape shuttle!




In all honesty if you could self replicate by laying chicken esque eggs every time they get fed, it'd be fun to either milk them for venom whilst in a pierce immune suit or just breed them for the purpose of dropping en-masse.

**Comment by Togopalis** [comment]
The placeholder sprite is currently being updated by Armhulenn!

**Comment by epicmemename** [comment]
no step on snek

**Comment by Anonmare** [comment]
Don't tread on me

File: code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: You don't need the else, just have it return ..() on same indentation as the if

**Comment by ExcessiveUseOfCobblestone** [comment]
hahahaha those sprites

File: code/modules/mob/living/simple_animal/friendly/snake.dm [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+	response_disarm = "shoos"
+	response_harm   = "steps on"
+	ventcrawler = VENTCRAWLER_ALWAYS
+	density = FALSE
+	pass_flags = PASSTABLE | PASSMOB
+	mob_size = MOB_SIZE_SMALL
+	gold_core_spawnable = FRIENDLY_SPAWN
+	obj_damage = 0
+	environment_smash = ENVIRONMENT_SMASH_NONE
+	var/static/list/edibles = typecacheof(list(/mob/living/simple_animal/butterfly, /mob/living/simple_animal/cockroach, /mob/living/simple_animal/mouse))
```
Comment (kevinz000) [inline code/modules/mob/living/simple_animal/friendly/snake.dm]: inb4 absolute madmin adds humans to it

Comment (Armhulen) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: you mean you

citadel man

File: code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 1 above)
Comment (Armhulen) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: noice

File: code/modules/mob/living/simple_animal/friendly/snake.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: wow.

Comment (FantasticFwoosh) [reply code/modules/mob/living/simple_animal/friendly/snake.dm]: Save it for giant pythons or something.

**Comment by Anonmare** [comment]
Irradiating a snake should have a chance of making a giant maneater snake that prowls maintenance - striking and constricting hapless greysuits before they can scream more than ";help maint"

**Comment by Armhulen** [comment]
can do once this is merged

**Comment by Armhulen** [comment]
but right now we're facing a conflict and losing 1 sec

**Comment by XDTM** [comment]
yuan-ti race when

**Comment by Togopalis** [comment]
Having problems modifying this PR, gonna have to make a newo ne

**Comment by Armhulen** [comment]
(we lost to the conflict)
:white_flag:

**Comment by Supermichael777** [comment]
Jesus christ
image

**Comment by Supermichael777** [comment]
Let me fix that

**Comment by Togopalis** [comment]
It was already fixed by Armhulenn thanks for offering for your help though!

**Comment by Anonmare** [comment]
no snek has been stepped

**Comment by Supermichael777** [comment]
prototype of a replacement @Togopalis  
image

**Pull request closed without merging**